drivers/pwm/pwm_nrfx.c

```diff
@@ -343,9 +353,18 @@ static int pwm_nrfx_pm_action(const struct device *dev,
 		.seq.length = NRF_PWM_CHANNEL_COUNT,			      \
 		.pcfg = PINCTRL_DT_DEV_CONFIG_GET(PWM(idx)),		      \
 	};								      \
+	static int pwm_nrfx_init##idx(const struct device *dev)		      \
+	{								      \
+		IF_ENABLED(NRFX_PWM_NRF52_ANOMALY_109_WORKAROUND_ENABLED,     \
+			(IRQ_CONNECT(DT_IRQN(PWM(idx)),			      \
+				     DT_IRQ(PWM(idx), priority),	      \
+				     nrfx_isr, nrfx_pwm_##idx##_irq_handler,  \
+				     0);))				      \
```

Comment:
I need to use the preprocessor in both cases:
- here, because the compiler would not optimize out `nrfx_pwm_*_irq_handler` functions if the workaround is not used (interrupt handlers are installed via a linker section, so even when the `if (0)` block is removed, those functions would stay referenced, resulting in some wasted flash)
- above, because the block could not be compiled when the `NRFX_PWM_NRF52_ANOMALY_109_EGU_INSTANCE` symbol is not defined (and it won't be for SoCs other then nRF52832)

so an `if` is not very suitable. I considered `#if (IS_ENABLED())` but I wasn't convinced that it would look much better (and it could not be used inside the macro so I'd need to define a separate one conditionally).

And I prefer to use the `NRFX_PWM_NRF52_ANOMALY_109_WORKAROUND_ENABLED` symbol instead of `CONFIG_NRF52_ANOMALY_109_WORKAROUND` because it can be potentially set without that Kconfig option enabled and that's what is actually used in the nrfx_pwm driver, so it's easier to find the relevant part of code there.

Reply:
gotcha, throwing more ideas: how about having a `#ifdef NRFX_PWM_NRF52_ANOMALY_109_WORKAROUND_ENABLED` that controls an alias of `IRQ_CONNECT` that goes to a no-op in the else case?

like
```c
#ifdef NRFX_PWM_NRF52_ANOMALY_109_WORKAROUND_ENABLED
extern void nrfx_egu_##idx##_irq_handler(void);
#define ANOMALY_IRQ_CONNECT(...) IRQ_CONNECT(...)
#else
#define ANOMALY_IRQ_CONNECT(...)
#endif
```

then the condition would be there only once and out of the code and in the normal case it should optimize fine.

Reply:
Updated using the above approach. Please take one more look.

Reply:
Thanks!